tests/lib/rules/file-extension-in-import.js

```diff
@@ -45,6 +45,11 @@ new RuleTester({
             filename: fixture("test.js"),
             code: "import '@typescript-eslint/parser'",
         },
+        {
+            filename: fixture("test.ts"),
+            code: "import {test} from '@typescript-eslint/parser/index'",
+            options: ["always"],
+        },
```

Comment:
if the `./index` was defined in its exports, the ext is disallowed, otherwise it's always required in esm.

```json
"exports": {"./index": "...."}
```